tests/test_httpserver.py

```diff
@@ -72,3 +72,58 @@ def test_limited_length_file_tell_on_socket():
     assert f.read() == b'123456789'
     assert f.tell() == 10
     backing_read.close()
+
+
+def test_address_family_v4():
+    #ipv4
+    app = None
+    host = '127.0.0.1'
+    port = '9090'
+    
+    svr = serve(app, host=host, port=port, start_loop=False, use_threadpool=False)
+
+    af = svr.address_family
+    addr = svr.server_address
+    p = svr.server_port
+    
+    svr.server_close()
+
+    assert (af == socket.AF_INET)
+    assert (addr[0] == '127.0.0.1')
+    assert (str(p) == port)
+
+
+def test_address_family_v4_host_and_port():
+    #ipv4
+    app = None
+    host = '127.0.0.1:9091'
+    
+    svr = serve(app, host=host, start_loop=False, use_threadpool=False)
+    
+    af = svr.address_family
+    addr = svr.server_address
+    p = svr.server_port
+    
+    svr.server_close()
+    
+    assert (af == socket.AF_INET)
+    assert (addr[0] == '127.0.0.1')
+    assert (str(p) == '9091')
+
+def test_address_family_v6():
+    #ipv6
+    app = None
+    host = '[::1]'
+    port = '9090'
+    
+    svr = serve(app, host=host, port=port, start_loop=False, use_threadpool=False)
```

Comment:
Weird, I was hoping that `start_loop=False` was going to be the fix for the test problem, but clearly not.

Another option would be to use the `skipnetwork` decorator used elsewhere in the tests on this test.

Reply:
Hmmn i couldnt find any skipnetwork decorator. Could you provide a reference to the specific file?

The alternative im exploring is to basically pass the test based on the OSError being raised for the ipV6 case, this means it will pass for TravisCI setup but probably fail if you run it locally.. or i could have logic to try/except that so it works in both.

Reply:
> Hmmn i couldnt find any skipnetwork decorator.

I'm sorry, I'm managing too many of these old-ish python modules. That decorator is from a different package (wsgi-intercept) and has enough setup involved that it would not be useful to add it into paste.

Your idea of having a try/except handling to work in different environments would probably work okay. Another option would be to figure out if travis exposes an environment variable that you can check for and either skip the test or adjust its behavior based on that.

And yet another option would be to extract the is_ipv6 processing to its own method and test just that instead of the server method. In fact that may be the best option.

But whichever you find most useful.

Thanks very much for taking this on.

Reply:
seems to be ok now with the exception handling logic.

Thanks for taking on maintaining the library!

Reply:
lgtm thanks!